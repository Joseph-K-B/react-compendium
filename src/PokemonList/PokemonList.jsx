import Pokemon from "../Pokemon/Pokemon";
import './PokemonList.css'

function PokemonList({ pokemons }) {
    return (
        <section className='pokemonList' aria-label='pokemonList'>
            {pokemons.map((pokemon) => {
                return (
                    <div key={pokemon.id} aria-label='pokemon'>
                        <Pokemon pokemon={pokemon}/>
                    </div>
                )

            })}
        </section>
    )
}

export default PokemonList