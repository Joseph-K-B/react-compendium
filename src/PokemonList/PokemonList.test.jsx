import PokemonList from "./PokemonList";
import { render, screen } from '@testing-library/react';

it('should render unordered list of pokemon', () => {
    const {container} = render(<PokemonList pokemons={['pikachu', 'charizard']}/>);
    const pokemonList = screen.getByLabelText(/pokemonList/);

    expect(pokemonList).toBeInTheDocument();
    expect(container).toMatchSnapshot();
})