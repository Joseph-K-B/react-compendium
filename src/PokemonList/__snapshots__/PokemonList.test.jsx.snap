// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render unordered list of pokemon 1`] = `
<div>
  <section
    aria-label="pokemonList"
    class="pokemonList"
  >
    <div
      aria-label="pokemon"
    >
      <figure
        aria-label="pokemonFigure"
        class="pokemonFigure"
      >
        <h2 />
        <img
          aria-label="pokemonName"
        />
        <h3 />
        <ul>
          <li>
            Primary type:
          </li>
          <li>
            Secondary type:
          </li>
          <li>
            abilitiy:
          </li>
          <li>
            ability:
          </li>
        </ul>
      </figure>
    </div>
    <div
      aria-label="pokemon"
    >
      <figure
        aria-label="pokemonFigure"
        class="pokemonFigure"
      >
        <h2 />
        <img
          aria-label="pokemonName"
        />
        <h3 />
        <ul>
          <li>
            Primary type:
          </li>
          <li>
            Secondary type:
          </li>
          <li>
            abilitiy:
          </li>
          <li>
            ability:
          </li>
        </ul>
      </figure>
    </div>
  </section>
</div>
`;
